extensions/sql-database-projects/src/tools/buildHelper.ts

```diff
@@ -67,66 +69,114 @@ export class BuildHelper {
 		const fullSdkName = `${sdkName}.${sdkVersion}`;
 
 		// check if this if the nuget needs to be downloaded
-		const nugetPath = path.join(this.extensionBuildDir, `${fullSdkName}.nupkg`);
+		const buildSdkNugetPath = path.join(this.extensionBuildDir, `${fullSdkName}.nupkg`);
```

Comment:
this feels like it's begging to be refactored into a(nother) helper method, maybe something like

`ensureNugetPresence(nugetPackageName: string, expectedFiles: string[])`

that handles checking for missing files, downloads the files, and extracts them if missing.  I think that'd be easier to follow than interweaved logic for two different nugets.

Reply:
ok refactored it so it should be a little easier to follow now!